File: strings/mergestrings.java
package strings;

class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder ns = new StringBuilder(); //new string
        int i = 0;
        
        // Merge the characters alternately from both words
        while (i < word1.length() && i < word2.length()) {
            ns.append(word1.charAt(i));
            ns.append(word2.charAt(i));
            i++;
        }
        
        // If there are remaining characters in word1, append them
        while (i < word1.length()) {
            ns.append(word1.charAt(i));
            i++;
        }
        
        // If there are remaining characters in word2, append them
        while (i < word2.length()) {
            ns.append(word2.charAt(i));
            i++;
        }
        
        // Convert the StringBuilder to a String and return it
        return ns.toString();    
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        String word1 = "abc";
        String word2 = "pqrst";
        System.out.println("Merged string is: " + solution.mergeAlternately(word1, word2));
    }
}

File: strings/mergestrings.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        len1 = len(word1)
        len2 = len(word2)
        newword = ''
        for i in range(min(len1, len2)):
            newword += word1[i]
            newword += word2[i]
        
        newword += word1[len2:] if len1 > len2 else word2[len1:]
        return newword
    
solution = Solution()
word1 = 'abc'
word2 = 'pqrst'
print("merged string is:", solution.mergeAlternately(word1, word2))
